<?xml version="1.0"?>

<launch>
    
   <arg name="robotname" default="robot_0" />
    
     <group ns="$(arg robotname)">      
    
        <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" >	 
                <rosparam file="$(find utils)/launch_robotcraft/costmap_common_params.yaml" command="load" ns="global_costmap" />
                <rosparam file="$(find utils)/launch_robotcraft/costmap_common_params.yaml" command="load" ns="local_costmap" />
                <rosparam file="$(find utils)/launch_robotcraft/local_costmap_params.yaml" command="load" />
                <rosparam file="$(find stopy_nav)/config_files/global_costmap_params.yaml" command="load" />	
                <rosparam file="$(find stopy_nav)/config_files/move_base_and_local_planner_params.yaml" command="load" />
                
                <remap from="map" to="/map"/>
                
                <!-- Override MOVE_BASE Frame/Topic Params from .yaml files to include prefix: -->
                <param name="global_costmap/laser_scan_sensor/sensor_frame" value="$(arg robotname)/base_laser_link"/>
                <param name="global_costmap/robot_base_frame" value="$(arg robotname)/base_link"/>   
                <param name="local_costmap/laser_scan_sensor/sensor_frame" value="$(arg robotname)/base_laser_link"/>              
                <param name="local_costmap/robot_base_frame" value="$(arg robotname)/base_link"/>             
                <param name="local_costmap/global_frame" value="$(arg robotname)/odom"/>
        </node>
    
    </group>
    
      
    <!-- Run rviz -->
    <node pkg="rviz" type="rviz" name="rviz_0" args="-d $(find utils)/launch_robotcraft/robotcraft.rviz" output="screen"/>
        
  
</launch>